drivers/adc/ad4858/ad4858.c

```diff
@@ -0,0 +1,1069 @@
+ * Copyright 2023(c) Analog Devices, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *  - Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *  - Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ *  - Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <string.h>
+#include "ad4858.h"
+#include "no_os_delay.h"
+#include "no_os_print_log.h"
+
+/**
+ * @brief Write device register.
+ * @param dev- The device structure.
+ * @param reg_addr - The register address.
+ * @param reg_val - The data to be written.
+ * @return 0 in case of success, negative error code otherwise.
+ * @note Multibyte write with streaming mode is supported by default.
+ */
+int ad4858_reg_write(struct ad4858_dev *dev, uint32_t reg_addr,
+		     uint32_t reg_val)
+{
+	int32_t ret;
+	uint32_t addr;
+	uint8_t cnt = 0;
+	uint8_t reg_size = AD4858_LEN(reg_addr);
+	uint8_t buf[6] = { 0 };
+
+	if (!dev)
+		return -EINVAL;
+
+	if (!dev->addr_ascension_enable) {
+		addr = AD4858_ADDR(reg_addr) + reg_size - 1;
+	} else {
+		addr = AD4858_ADDR(reg_addr);
+	}
+
+	buf[cnt++] = (uint8_t)(addr >> 8);
```

Comment:
use no_os_field_get

Reply:
do you mean no_os_put_unaligned_be16? This is the only relevant function I see.
However, the problem with this function is, it's name indicates "be- big endian" but my processor is little endian (as most ARM processors are LE). So this might create some confusion.